base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/table/ActionsTable.java

```diff
@@ -162,30 +160,45 @@ public void clearAction() {
         final List<ScriptGeneratorAction> newList = new ArrayList<ScriptGeneratorAction>();
         firePropertyChange(ACTIONS_PROPERTY, actions, actions = newList);
     }
+	
+	/**
+	 * Moves a selection of actions down one row.
+	 * @param actionsToMove
+	 * 			The actions to be moved
+	 */
+	public void moveActionDown(List<ScriptGeneratorAction> actionsToMove) {
+		if (actionsToMove.contains(actions.get(actions.size() - 1))) {
+			return;
+		}
+		final var newActions = new ArrayList<ScriptGeneratorAction>(actions);
+		ListIterator<ScriptGeneratorAction> iterator = actionsToMove.listIterator(actionsToMove.size());
 
+		// Iterate in reverse.
+		while (iterator.hasPrevious()) {
+			Integer currentIndex = actions.indexOf(iterator.previous());
+			Collections.swap(newActions, currentIndex, currentIndex + 1);
+		}
+		
+		firePropertyChange(ACTIONS_PROPERTY, actions, actions = newActions);
+	}
+	
 	/**
-	 * Moves action to a new position in the table.
-	 * @param oldIndex
-	 * 			The current index of the action to be moved.
-	 * @param newIndex
-	 * 			The index to move the action to, if valid.
+	 * Moves a selection of actions up a row.
+	 * @param actionsToMove
+	 * 			The indices of actions to be moved
 	 */
-	public void moveAction(int oldIndex, int newIndex) {
-		if (isValidIndex(oldIndex)) {
-			final var newActions = new ArrayList<ScriptGeneratorAction>();
-			
-			newActions.addAll(actions);
-			
-			if (newIndex < 0) {
-				newIndex = 0;
-			} else if (newIndex >= this.actions.size()) {
-				newIndex = this.actions.size() - 1;
-			}
-			
-			Collections.swap(newActions, oldIndex, newIndex);
+	public void moveActionUp(List<ScriptGeneratorAction> actionsToMove) {
+		if (actionsToMove.contains(actions.get(0))) {
```

Comment:
I get the following error if I have no actions in the table and press the move up/mode down buttons. I think we should fix this by greying out the buttons if there are no actions in the table/no actions selected.

```
*2020-06-19 14:37:55.819 [main] ERROR org.eclipse.ui.workbench - Index 0 out of bounds for length 0
	java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	java.base/java.util.Objects.checkIndex(Objects.java:372)
	java.base/java.util.ArrayList.get(ArrayList.java:458)
	uk.ac.stfc.isis.ibex.scriptgenerator.table.ActionsTable.moveActionUp(ActionsTable.java:191)
	uk.ac.stfc.isis.ibex.scriptgenerator.ScriptGeneratorSingleton.moveActionUp(ScriptGeneratorSingleton.java:473)
	uk.ac.stfc.isis.ibex.ui.scriptgenerator.views.ScriptGeneratorViewModel.moveActionUp(ScriptGeneratorViewModel.java:323)
	uk.ac.stfc.isis.ibex.ui.scriptgenerator.views.ScriptGeneratorView.lambda$8(ScriptGeneratorView.java:297)
	org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	org.eclipse.swt.widgets.Display.sendEvent(Display.java:4175)
	org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1057)
	org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3988)
	org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3587)
	org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1160)
	org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1049)
	org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:660)
	org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:154)
	uk.ac.stfc.isis.ibex.scriptgenerator.product.Application.start(Application.java:47)
	org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.base/java.lang.reflect.Method.invoke(Method.java:566)
	org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:657)
	org.eclipse.equinox.launcher.Main.basicRun(Main.java:594)
	org.eclipse.equinox.launcher.Main.run(Main.java:1465)
	org.eclipse.equinox.launcher.Main.main(Main.java:1438)

```